hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -495,12 +495,27 @@ public boolean isVolumeEmpty(String volume) throws IOException {
   public boolean isBucketEmpty(String volume, String bucket)
       throws IOException {
     String keyPrefix = getBucketKey(volume, bucket);
-    //TODO: When Key ops are converted in to HA model, use cache also to
-    // determine bucket is empty or not.
+
+    // First check in key table cache.
+    Iterator<Map.Entry<CacheKey<String>, CacheValue<OmKeyInfo>>> iterator =
+        ((TypedTable< String, OmKeyInfo>) keyTable).cacheIterator();
+    while (iterator.hasNext()) {
+      Map.Entry< CacheKey<String>, CacheValue<OmKeyInfo>> entry =
+          iterator.next();
+      String key = entry.getKey().getCacheKey();
+      OmKeyInfo omKeyInfo = entry.getValue().getCacheValue();
+      // Making sure that entry is not for delete key request.
+      if (key.startsWith(keyPrefix) && omKeyInfo != null) {
+        return false;
+      }
+    }
     try (TableIterator<String, ? extends KeyValue<String, OmKeyInfo>> keyIter =
         keyTable.iterator()) {
       KeyValue<String, OmKeyInfo> kv = keyIter.seek(keyPrefix);
-      if (kv != null && kv.getKey().startsWith(keyPrefix)) {
+      // During iteration from DB, check in mean time if this key is not
+      // marked for delete.
+      if (kv != null && kv.getKey().startsWith(keyPrefix) &&
+          keyTable.get(kv.getKey()) != null) {
```

Comment:
Done

Reply:
Actually thinking more, we don't need 2nd check at all.

Because to delete Bucket, we acquire Bucket lock, and then check isBucketEmpty. So during this time, no operations can be added to the cache.

So in isBucketEmpty, first, check the cache, and then next DB. I feel that should be sufficient. Let me know if i am missing something here.

Reply:
We would have to check the cache to not count the keys which are marked for delete.

Reply:
Yes, you are correct. I have missed this part. 
Updated the code to fix this.